src/sorted-table.js

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import {useState} from "react";
+import "./sorted-table.css";
+
+export default function SortedTable({tableData}) {
+
+    const tableContentWithKey = () => tableData.slice(1).map((row, i) => [i].concat(row));
+    const compare = (a,b) =>
+        typeof a === "number" ? a - b
+            : typeof a === "string" ? (a.toLowerCase() < b.toLowerCase() ? -1 : a.toLowerCase() > b.toLowerCase() ? 1 : 0)
+            : typeof a === "boolean" ? (a && !b ? 1 : a === b ? 0 : -1)
+            : 0;
+
+    const [sortColumn, setSortColumn] = useState(0);
+    const [tableContent, setTableContent] = useState(tableContentWithKey());
+    const [ascending, setAscending] = useState(true);
+
+
+
+    function sortTable(tbl, column, asc) {
+        return tbl.map((row, i) => [row[column], i])
+            .sort((a, b) => compare(a[0], b[0]) * (asc ? 1 : -1))
+            .map(a => tbl[a[1]]);
+    }
+
+    function onSortClick(columnIndex) {
+        const asc = columnIndex !== sortColumn || !ascending;
+        setSortColumn(() => columnIndex);
+        setTableContent(sortTable(tableContent, columnIndex, asc));
+        setAscending(() => asc);
```

Comment:
Why are you using a function here?

Reply:
According to Ynon, it is better to provide to provide a function and not the actual value.
Let me know if there's a refinement to that guideline

Reply:
Just saw this - the guideline is: 

1. Use a function when the change *uses* the previous value (for example increase a value by one)

2. Use a fixed value when the change is not about changing the previous value (for example "reset" which sets always to 0)